Mathlib/Data/Fin/Tuple/AppendLemmas.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2025 Bjørn Solheim. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Solheim
+-/
+import Mathlib.Data.Fin.Tuple.Basic
+
+/-!
+# Appending finite sequences
+
+We give various basic results regarding multiplying, appending,
+and splitting finite sequences.
+
+## Main results
+
+- A basic API for appending and de-appending finite sequences
+
+sample lemmas:
+-'append_apply_lt': extracting the initial part of the append of two sequences
+-'append_apply_ge': extracting the terminal part of the append of two sequences
+-'append_forall_iff': appended sequences preserve predicates of its components
+
+## Notation
+
+ - no special notation defined
+
+## Implementation notes
+
+In this file finite sequences are represented by
+  'v : Fin n → E' (representing vectors or module elements) and
+  'c : Fin m → R' (representing scalar coefficients).
+
+## References
+
+ - This way of defining finite sequences is used in
+  Mathlib.Data.Fin.Tuple.Basic
+
+-/
+variable {m n : ℕ}
+variable {α : Sort*} {β : Type*} {γ : Fin (m + n) → Sort*}
+
+namespace Fin
+
+/-- Appending two sequences v and w, and then extracting
+the less than part, gives back the original first sequence v. -/
+theorem append_apply_lt (v : Fin m → α) (w : Fin n → α) (i : Fin (m + n)) (h : ↑i < m) :
+    append v w i = v ⟨↑i, h⟩ := by
+  rw [← append_left v w]
+  rfl
+
+/-- Appending two sequences v and w, and then extracting
+the greater than or equal part, gives back the original second sequence w. -/
+theorem append_apply_ge (v : Fin m → α) (w : Fin n → α) (i : Fin (m + n)) (h : m ≤ ↑i) :
+    append v w i = w ⟨↑i - m, Nat.sub_lt_left_of_lt_add h i.isLt⟩ := by
+  rw [← append_right v w]
+  congr!
+  exact Fin.ext (Nat.add_sub_of_le h).symm
+
+/-- A finite sequence of properties P holds for {0 , ... , m + n - 1} iff
+it holds separately for both {0 , ... , m - 1} and {m, ..., m + n - 1}. -/
+theorem forall_fin_add {m n} (P : Fin (m + n) → Prop) :
+    (∀ i, P i) ↔ (∀ i, P (castAdd _ i)) ∧ (∀ i, P (natAdd _ i)) :=
+  ⟨fun h => ⟨fun _ => h _, fun _ => h _⟩, fun ⟨hm, hn⟩ => Fin.addCases hm hn⟩
+
+/-- A property P holds for all elements of an appended sequence iff
+it holds for all elements of both original sequences. -/
+theorem append_forall_iff (P : α → Prop) (v1 : Fin m → α) (v2 : Fin n → α) :
```

Comment:
I think I'd suggest dropping this lemma, unless we have precedent for it in the form of the `cons` version already existing.